### ISSUE TITLE
Git Clone Optimization 

### ISSUE THREAD

**Issue by bkottcamp** [davidahouse/stampede-worker#206]
Github is recommending partial clone instead of shallow clone. See:
https://github.blog/2020-12-21-get-up-to-speed-with-partial-clone-and-shallow-clone

Which says among other things that shallow clones are strongly discouraged (emphasis added):
git clone --depth=1 <url> creates a shallow clone. These clones truncate the commit history to reduce the clone size. This creates some unexpected behavior issues, limiting which Git commands are possible. These clones also put undue stress on later fetches, so they are **strongly discouraged** for developer use. They are helpful for some build environments where the repository will be deleted after a single build.

One example of "creates some unexpected behavior issues" is when Sonarqube is unable to correctly identify blame.

The recommendation is to use:
git clone --filter=tree:0 <url>

**Comment by bkottcamp** [comment]
@davidahouse what do you think about using filter tree instead of shallow clone?

**Comment by davidahouse** [comment]
@bkottcamp Yeah it looks like filter is a good choice if you need to run further git commands as it will fetch what it needs on demand. Looking back now I have no idea why I didn't make the entire clone options configurable. 😵‍💫 

I can create a version with it configurable (but I'll leave the default as shallow clone).
